perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/pwdmgr/GenericPasswordManagerModule.java

```diff
@@ -55,64 +64,65 @@ public void setActualLoginNamespace(String actualLoginNamespace) {
 	}
 
 	@Override
-	public Map<String, String> generateAccount(PerunSession session, Map<String, String> parameters) throws InternalErrorException {
+	public Map<String, String> generateAccount(PerunSession sess, Map<String, String> parameters) {
 		// account generation is not supported
 		return null;
 	}
 
 	@Override
-	public void reservePassword(PerunSession session, String userLogin, String password) throws InternalErrorException {
-		if (StringUtils.isBlank(password)) {
-			throw new EmptyPasswordRuntimeException("Password for " + actualLoginNamespace + ":" + userLogin + " cannot be empty.");
-		}
+	public void reservePassword(PerunSession sess, String userLogin, String password) throws InvalidLoginException, PasswordStrengthException {
+		checkLoginFormat(sess, userLogin);
+		checkPasswordStrength(sess, userLogin, password);
 		Process process = createPwdManagerProcess(PASSWORD_RESERVE, actualLoginNamespace, userLogin);
 		sendPassword(process, password);
 		handleExit(process, actualLoginNamespace, userLogin);
 	}
 
 	@Override
-	public void reserveRandomPassword(PerunSession session, String userLogin) throws InternalErrorException {
+	public void reserveRandomPassword(PerunSession sess, String userLogin) throws InvalidLoginException {
+		checkLoginFormat(sess, userLogin);
 		Process process = createPwdManagerProcess(PASSWORD_RESERVE_RANDOM, actualLoginNamespace, userLogin);
 		handleExit(process, actualLoginNamespace, userLogin);
 	}
 
 	@Override
 	public void checkPassword(PerunSession sess, String userLogin, String password) {
+		// use custom check instead of strength, since this is about empty input
```

Comment:
I am not sure what does this comment mean.

Reply:
I rephrase that. Basically we must not check password strength here, since we check "old" possibly weaker password. We just must make sure, that input is not empty.